ACE/ace/DLL_Manager.cpp

```diff
@@ -241,15 +241,15 @@ ACE_DLL_Handle::open (const ACE_TCHAR *dll_name,
               if (errors || ACE::debug ())
                 {
                   ACE_TString errtmp;
-                  if (errors)
+                  this->error (errtmp);
+
+                  if (errors && errors->is_empty())
```

Comment:
Why an added check for is_empty here?

Reply:
Can these lines be removed? It looks like duplicated.

Reply:
Can it be that error==0 but debug==1?

Reply:
No matter whether `errors` is 0, `errtmp` is set by `this->error` before.

But after reading it again just now, I found line from 239 to 262 can be removed because all errors are pushd or logging inside while loop before.

Am I right?

Reply:
Not sure, would keep them for the moment

Reply:
OK. Given keeping them, 

If `is_empty` is true, no error messages are **pushed** into `errors` and **logged**.

Otherwise error messages are **pushed** and **logged** so that it is not necessary to push and log it again.